galleon-pack/src/main/resources/feature_groups/jca.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature-group-spec name="jca" xmlns="urn:jboss:galleon:feature-group:1.0">
+    <feature spec="subsystem.jca">
+        <feature spec="subsystem.jca.archive-validation.archive-validation">
+            <param name="enabled" value="true"/>
+            <param name="fail-on-error" value="true"/>
+            <param name="fail-on-warn" value="false"/>
+        </feature>
+        <feature spec="subsystem.jca.bean-validation.bean-validation">
+            <param name="enabled" value="true"/>
+        </feature>
+        <feature spec="subsystem.jca.cached-connection-manager.cached-connection-manager">
+            <param name="install" value="true" />
+        </feature>
+        <feature spec="subsystem.jca.workmanager">
+            <param name="workmanager" value="default" />
+            <param name="name" value="default" />
+            <feature spec="subsystem.jca.workmanager.long-running-threads">
+                <param name="long-running-threads" value="default"/>
+                <param name="max-threads" value="50"/>
+                <param name="queue-length" value="50"/>
+                <param name="core-threads" value="50"/>
+                <param name="keepalive-time" value="{time=10,unit=SECONDS}"/>
+            </feature>
+            <feature spec="subsystem.jca.workmanager.short-running-threads">
+                <param name="short-running-threads" value="default"/>
+                <param name="max-threads" value="50"/>
+                <param name="queue-length" value="50"/>
+                <param name="core-threads" value="50"/>
+                <param name="keepalive-time" value="{time=10,unit=SECONDS}"/>
+            </feature>
```

Comment:
This feature should come before long-running-threads.

There's a bug in the subsystem persister in that it doesn't persist in xsd order, if it did this wouldn't matter, but since it doesn't getting these features in the right order is an easy way to avoid diffs.

Reply:
Persisting in the valid XSD order could also enable potential feature arranging optimizations which would otherwise break the validity of the resulting XML file.

Reply:
It's definitely the rule that persistence should happen in xsd order. Anywhere you see that not happening please file a JIRA. https://issues.jboss.org/browse/WFLY-10347 is for this issue.

It's tough to build in behavior absolutely depending on there not being bugs like this though. At least not with the amount of custom marshalling code we have. We require our standard configs that we've hand coded to be in xsd order, which means the inputs to the system are in that order, and therefore we're not providing 'difficult inputs' that would expose bugs.